src/mono/mono/metadata/reflection.c

```diff
@@ -963,7 +963,7 @@ add_parameter_object_to_array (MonoDomain *domain, MonoMethod *method, MonoObjec
 
 	MonoObjectHandle def_value;
 
-	if (!(sig_param->attrs & PARAM_ATTRIBUTE_HAS_DEFAULT)) {
+	if (!(sig_param->attrs & PARAM_ATTRIBUTE_HAS_DEFAULT) || method->wrapper_type != MONO_WRAPPER_NONE) {
 		if (sig_param->attrs & PARAM_ATTRIBUTE_OPTIONAL)
```

Comment:
Note that dynamic methods also have wrapper_type set.

Reply:
Added a check for `MONO_WRAPPER_TYPE_DYNAMIC_METHOD`.  Verified that we don't have a crash when printing a stack trace from inside a dynamic method